core/src/Slay/Prim.hs

```diff
@@ -82,6 +86,18 @@ data PrimCurve g =
       curveDirection :: g Direction
     }
 
+
+-- | circle adt
+data PrimCircle g = Circle
+    { circleColor :: g (Maybe Color)
+    , circleCenterX :: Double
```

Comment:
I called them like that since they are X and Y  positions of the center of the arc, if you got a better name i will be glad to change it.